StreetSmartAPI/StreetSmart.WPF.Tests/StreetSmartAPITests.cs

```diff
@@ -131,4 +133,149 @@ public void CloseViewer_Empty()
     Mock.Verify();
   }
 #endif
+  [Fact]
+  public async Task AddOverlay_ValidJson_ReturnsOverlay()
+  {
+    var geoJsonOverlay = new GeoJsonOverlay("{\"type\":\"FeatureCollection\",\"features\":[]}", "myName", "mySrs", Color.AliceBlue);
+    _browserMock.Setup(x => x.IsDisposed).Returns(false);
+    _browserMock.Setup(x => x.GetBrowser()).Returns(_iBrowserMock.Object);
+    _browserMock.Setup(x => x.ExecuteScriptAsync(It.IsAny<string>())).Callback(() => _viewer.OnResult(new { id = 4 }, "AddOverlay1")).Verifiable(Times.Once);
+
+    var result = await _viewer.AddOverlay(geoJsonOverlay);
+
+    Assert.Equal(geoJsonOverlay, result);
+    Mock.Verify();
+  }
+
+  [Fact]
+  public async Task AddOverlay_InvalidJson_ThrowsStreetSmartJsonException()
+  {
+    var invalidJsonOverlay = new GeoJsonOverlay("{invalid json}", "myName", "mySrs", Color.AliceBlue);
+
+    await Assert.ThrowsAsync<StreetSmart.Common.Exceptions.StreetSmartJsonException>(() => _viewer.AddOverlay(invalidJsonOverlay));
+  }
+
+  [Fact]
+  public async Task AddOverlay_NullOverlay_ThrowsStreetSmartJsonException()
+  {
+    var nullJsonOverlay = new GeoJsonOverlay(null, "myName", "mySrs", Color.AliceBlue);
+
+    await Assert.ThrowsAsync<StreetSmart.Common.Exceptions.StreetSmartJsonException>(() => _viewer.AddOverlay(nullJsonOverlay));
+  }
+
+  [Theory]
+  [InlineData("{}", true)]
+  [InlineData("[]", true)]
+  [InlineData("{\"key\":\"value\"}", true)]
+  [InlineData("[{\"key\":\"value\"}]", true)]
+  [InlineData("", false)]
+  [InlineData(" ", false)]
+  [InlineData("{invalid}", false)]
+  [InlineData("[{invalid}]", false)]
+  [InlineData("{\"key\": \"value\", \"key2\": {\"nestedKey\": \"nestedValue\"}}", true)]
+  [InlineData("{\"key\": \"value\", \"key2\": {\"nestedKey\": undefined}}", false)]
+  public void IsValidJson_ValidatesJsonStringsCorrectly(string input, bool expectedResult)
+  {
+    var result = input.IsValidJson();
+
+    Assert.Equal(expectedResult, result);
+  }
+
+  [Fact]
+  public void IsValidJson_WhitespaceInput_ReturnsFalse()
+  {
+    string input = "   ";
+
+    var result = input.IsValidJson();
+
+    Assert.False(result);
+  }
+
+  [Fact]
+  public void IsValidJson_InvalidJson_ReturnsFalse()
+  {
+    string input = "{invalid json}";
+
+    var result = input.IsValidJson();
+
+    Assert.False(result);
+  }
+
+  [Fact]
+  public void IsValidJson_ValidJsonWithComplexStructure_ReturnsTrue()
```

Comment:
I think this can be moved to test IsValidJson_ValidatesJsonStringsCorrectly, since we already have lot of InlineData defined in that test.